blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunSearch.java

```diff
@@ -83,6 +84,40 @@ public Link getLink() {
         return runs;
     }
 
+    public static Iterable<BlueRun> findRuns(Job job, final Link parent, int start, int limit){
+        final List<BlueRun> runs = new ArrayList<>();
+        Iterable<Job> pipelines;
+        if(job != null){
+            pipelines = ImmutableList.of(job);
+        }else{
+            pipelines = Jenkins.getInstance().getItems(Job.class);
+        }
+
+        int skipCount=0;
+        int limitCount=0;
+        for (Job p : pipelines) {
+            RunList<? extends Run> runList = p.getBuilds();
```

Comment:
@tfennelly getBuilds() returns a lazy list, though. We just can't enumerate it all, or it _will_ load them up.